legoMain.c

```diff
@@ -16,86 +16,121 @@ const int tapeIndicator = 3850;
+    	set_servo_position(sideServo, innerSide);
+    	set_servo_position(rotateServo, verticalArm);
+    	set_servo_position(clawServo, closedClaw);
+    	disable_servos();
 
     // Waiting for Start Light
-    wait_for_light(lightSensor);
-    shut_down_in(118);
+    	wait_for_light(lightSensor);
+    	shut_down_in(118);
     
     // Push 4 initial left side habitats to Area 1
-    enable_servos();
-    set_servo_position(sideServo, pushedSide);
-    msleep(50);
-    set_servo_position(sideServo, innerSide);
+    	enable_servos();
+    	set_servo_position(sideServo, pushedSide);
+    	msleep(50);
+    	set_servo_position(sideServo, innerSide);
 
     // Line up with tape
-    alignTape(250);
+    	alignTape(250);
 
     // Sweep top rocks to Area 5 Intersection
-    driveForward(75, leftVelocity-100, rightVelocity);
-    turnRight(80);
-    driveForward(18, leftVelocity, rightVelocity);
-    turnRight(30);
-    driveForward(10, leftVelocity, rightVelocity);
-    driveForward(20, -1 * leftVelocity, -1 * rightVelocity);
+    	driveForward(75, leftVelocity-100, rightVelocity);
+    	turnRight(80);
+    	driveForward(18, leftVelocity, rightVelocity);
+    	turnRight(30);
+    	driveForward(10, leftVelocity, rightVelocity);
+    	driveForward(20, -1 * leftVelocity, -1 * rightVelocity);
 
     // Red Poms
-    turnRight(60);
-    driveForward(20, leftVelocity, rightVelocity);
-    turnLeft(40);
+    	turnRight(60);
+    	driveForward(20, leftVelocity, rightVelocity);
+    	turnLeft(40);
 	driveForward(27, leftVelocity, rightVelocity);
-    driveForward(20, -1 * leftVelocity, -1 * rightVelocity);
-    alignTape(-250);
+    	driveForward(20, -1 * leftVelocity, -1 * rightVelocity);
+    	alignTape(-250);
     
     // Light Switch
-    turnLeft(90);
-    alignTape(500);
+    /*
+    	turnLeft(90);
+	alignTape(500);
 	driveForward(10, leftVelocity, rightVelocity);
-    turnLeft(90);
-
-    // not finished
-    cmpc(0);
-    set_servo_position(rotateServo, 1400);
-    while (analog(etSensor) < 2800) { 
-        driveForward(2, 100, 100);
-    }
-    set_servo_position(rotateServo, 1600);
-    msleep(400);
-    set_servo_position(rotateServo, 1400);
+    	turnLeft(90);
+    */
 
     // Top Cubes
-    driveForward(10, -500, -500)
-    set_servo_position(rotateServo, verticalArm);
+    	turnLeft(90);
+    	alignTape(500);
+	driveForward(12, leftVelocity, rightVelocity);
+    	turnLeft(95);
+    	msleep(500);
+    	set_servo_position(0, 1250);
+    	msleep(500);
+    	set_servo_position(0, 1210);
+    	msleep(500);
+    	set_servo_position(1, 1603);
+    	msleep(1000);
+    	driveForward(25, leftVelocity, rightVelocity);
+    	msleep(1000);
+    	set_servo_position(1, 1024);
+    	msleep(500);
+    	driveForward(10, -1 * leftVelocity, -1 * rightVelocity); // has grabb yey
+    
 
     
 
 }
 
+// Changes how fast the servo goes, otherwise we're gonna make the robot do a kickflip. 
+void servoSpeed(int port, int speed, int intialPosition, int endPosition) {
```

Comment:
Should you need more information on how this works, I will modify it to make it more condensed, but I will explain if need be.